solvers/irls.py

```diff
@@ -0,0 +1,53 @@
+from benchopt import BaseSolver, safe_import_context
+from benchopt.stopping_criterion import SufficientProgressCriterion
+
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from scipy import sparse
+
+
+class Solver(BaseSolver):
+    name = "IRLS"  # Iterative Reweighted Least Squares
+    stopping_criterion = SufficientProgressCriterion(
+        patience=6, strategy="iteration"
+    )
+    name = "IRLS"  # Iterative reweighted Least Squares
+    stopping_strategy = "iteration"
+
+    references = ["???"]
+    # see for instance: https://homepages.laas.fr/vmagron/masmode/Gabriel.pdf
+
+    def skip(self, X, y, lmbd, fit_intercept):
+        # XXX - not implemented but not too complicated to implement
+        if fit_intercept:
+            return True, f"{self.name} does not handle fit_intercept"
+
+        return False, None
+
+    def set_objective(self, X, y, lmbd, fit_intercept):
+        self.X, self.y, self.lmbd = X, y, lmbd
+        self.fit_intercept = fit_intercept
+
+    def run(self, n_iter):
+        epsilon = 1e-25
+
+        def w_opt(eta):
+            T = self.X.T @ self.X + self.lmbd * np.diag(1.0 / eta)
```

Comment:
You could use the dual formulation when more features than samples. Another option is to scale the X and just use the ridge_solver in sklearn

Reply:
This now added, thanks for the proposition (v1 was for having the original formula).